torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -478,6 +494,108 @@ def _low_precision_hook_enabled(state: _State) -> bool:
     return state._communication_hook in LOW_PRECISION_HOOKS
 
 
+@no_type_check
+@torch.no_grad()
```

Comment:
This used to be called `_wait_for_post_backward()`, but I found it has been easier to refer to this as the post-backward callback (hence why I renamed to `_post_backward_callback()`). It is registered as a callback that runs after the backward.